.github/workflows/main.yml

```diff
@@ -33,6 +33,32 @@ jobs:
     - name: Make distcheck
       run: make distcheck
 
+  build-usan:
+    name: Build with UCS2 and usan
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: sudo apt-get update -qq && sudo apt-get install -y libyaml-dev texinfo texlive
+    - name: Autogen && configure
+      run: |
+        ./autogen.sh
+        CFLAGS="-fsanitize=undefined -g" ./configure
+    - name: Make
+      run: make
+    - name: Make check
+      run: UBSAN_OPTIONS=halt_on_error=1 make check
+    - name: Store the test suite log
+      if: ${{ always() }} # store the test suite log even if the tests failed
+      uses: actions/upload-artifact@v2
+      with:
+        name: test-suite-ucs2.log
+        path: tests/test-suite.log
+    - name: Make distcheck
```

Comment:
I don't think we need to run `make distcheck` after we check for memory problems with `make check`